stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/thread/RepliesQueryReference.kt

```diff
@@ -0,0 +1,39 @@
+package io.getstream.chat.android.offline.experimental.channel.thread
+
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.call.Call
+import io.getstream.chat.android.client.call.await
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+import io.getstream.chat.android.core.internal.InternalStreamChatApi
+import io.getstream.chat.android.offline.experimental.channel.thread.state.ThreadState
+import io.getstream.chat.android.offline.experimental.extensions.state
+import io.getstream.chat.android.offline.experimental.plugin.QueryReference
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@ExperimentalStreamChatApi
+@InternalStreamChatApi
+/** Reference for the [ChatClient.getReplies] request. */
+public class RepliesQueryReference(
+    private val messageId: String,
+    private val limit: Int,
+    private val chatClient: ChatClient,
+) : QueryReference<List<Message>, ThreadState> {
+
+    /** Returns a call instance representing output of the [ChatClient.getReplies] request. */
+    override fun get(): Call<List<Message>> {
+        return chatClient.getReplies(messageId, limit)
+    }
+
+    /**
+     * Returns [ThreadState] for the thread replies of a message with id equal to [messageId]. And fill it by data from
+     * [ChatClient.getReplies].
+     *
+     * @param scope Coroutine scope where initial data filling action is being invoked.
+     */
+    override fun asState(scope: CoroutineScope): ThreadState {
+        scope.launch { get().await() }
```

Comment:
What happens with the list of messages here? I imagine they get updated somewhere internally, but to me it's super weird seeing a coroutine launched to get a list of data and then not doing anything with it.

Reply:
The logic of our offline support is so that we expose some state objects with reactive data and under the hood, we run a request to cache and BE to obtain data and fill those reactive sources with it. So `ThreadState` that is returned here provides you a possibility to bind your UI to reactive messages source.

Reply:
I understand it's updating data internall and that the ThreadState gives you the option of using the state.

However I don't understand why we launch a coroutine and await it, if the result value is never used (it just updates internal value).

Can we do a simple `enqueue()` here?

Reply:
Yes, we can `enqueue`. But passing scope you can have control flow of request lifecycle. I was thinking it's as use the full feature. If you think it's not needed we can remove it

Reply:
I think we can keep it - but just document it a bit more if possible.

E.g.

"We fetch the data in a coroutine and await its result because we want to update the internal data based on the result. We use a coroutine instead of a simple `enqueue()` call, to have control over the lifecycle of the request."

I think that should be good enough so that people know what's going on!

Reply:
But it says pretty the same, doesn't it?

`@param scope Coroutine scope where initial data filling action is being invoked.`

We can change `action` to `coroutine` for example. What do you think?

Reply:
Hmmm yeah it's mostly the same - I think `action` can stay.

I think maybe just adding: "We use a coroutine instead of a simple enqueue() call, to have control over the lifecycle of the request." would be helpful then so that we know why we chose one over the other.

If you feel like this doesn't make sense, we can just resolve this convo and move on with the merging process! :]

Reply:
For me your feedback is valuable. And I'm glad to find such description that make code reading easy